diff --git a/sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OfflineRecognizerResult.java b/sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OfflineRecognizerResult.java
index d763c32..ffdd658 100644
--- a/sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OfflineRecognizerResult.java
+++ b/sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OfflineRecognizerResult.java
@@ -12,0 +13 @@ public class OfflineRecognizerResult {
+    private final float[] ysLogProbs;
@@ -14 +15 @@ public class OfflineRecognizerResult {
-    public OfflineRecognizerResult(String text, String[] tokens, float[] timestamps, String lang, String emotion, String event, float[] durations) {
+    public OfflineRecognizerResult(String text, String[] tokens, float[] timestamps, String lang, String emotion, String event, float[] durations, float[] ysLogProbs) {
@@ -21,0 +23 @@ public class OfflineRecognizerResult {
+        this.ysLogProbs = ysLogProbs;
@@ -50,0 +53,4 @@ public class OfflineRecognizerResult {
+
+    public float[] getYsLogProbs() {
+        return ysLogProbs;
+    }
diff --git a/sherpa-onnx/jni/offline-recognizer.cc b/sherpa-onnx/jni/offline-recognizer.cc
index d160bc5..7ba9863 100644
--- a/sherpa-onnx/jni/offline-recognizer.cc
+++ b/sherpa-onnx/jni/offline-recognizer.cc
@@ -664 +664 @@ Java_com_k2fsa_sherpa_onnx_OfflineRecognizer_getResult(JNIEnv *env,
-                       "String;Ljava/lang/String;Ljava/lang/String;[F)V");
+                       "String;Ljava/lang/String;Ljava/lang/String;[F[F)V");
@@ -688,0 +689,3 @@ Java_com_k2fsa_sherpa_onnx_OfflineRecognizer_getResult(JNIEnv *env,
+  jfloatArray jysLogProbs = env->NewFloatArray(result.ys_log_probs.size());
+  env->SetFloatArrayRegion(jysLogProbs, 0, result.ys_log_probs.size(),
+                           result.ys_log_probs.data());
@@ -691 +694,2 @@ Java_com_k2fsa_sherpa_onnx_OfflineRecognizer_getResult(JNIEnv *env,
-                                   jlang, jemotion, jevent, jdurations);
+                                   jlang, jemotion, jevent, jdurations,
+                                   jysLogProbs);
@@ -699,0 +704 @@ Java_com_k2fsa_sherpa_onnx_OfflineRecognizer_getResult(JNIEnv *env,
+  env->DeleteLocalRef(jysLogProbs);
